front/pages/w/[wId]/assistant/new.tsx

```diff
@@ -173,185 +174,191 @@ export default function AssistantNew({
           />
         }
       >
-        {!conversation ? (
-          <div className="text-sm font-normal text-element-800">
-            <Page.Vertical gap="md" align="left">
-              <Page.Header
-                title={"Welcome " + user.name.split(" ")[0] + "!"} //Not solid
-                icon={ChatBubbleLeftRightIcon}
-              />
-              {/* FEATURED AGENTS */}
-              <Page.Vertical gap="lg" align="left">
-                <Page.Vertical gap="xs" align="left">
-                  <Page.SectionHeader title="Meet your team" />
-                  {isBuilder && (
-                    <>
-                      <Page.P variant="secondary">
-                        Dust comes with multiple assistants, each with a
-                        specific set of skills.
-                        <br />
-                        Create assistants tailored for your needs.
-                      </Page.P>
-                    </>
-                  )}
-                  {!isBuilder && (
-                    <>
-                      <Page.P variant="secondary">
-                        Dust is not just a single assistant, it’s a full team at
-                        your service.
-                        <br />
-                        Each member has a set of specific set skills.
-                      </Page.P>
-                      <Page.P variant="secondary">
-                        Meet some of your assistants team:
-                      </Page.P>
-                    </>
-                  )}
-                </Page.Vertical>
-                <div className="flex flex-col gap-2">
-                  <div className="grid grid-cols-2 gap-4 sm:grid-cols-4">
-                    {displayedAgents.map((agent) => (
-                      <a
-                        key={agent.sId}
-                        className="cursor-pointer"
-                        onClick={() => {
-                          void handleSubmit(
-                            `Hi :mention[${agent.name}]{sId=${agent.sId}}, what can you help me with?`,
-                            [
-                              {
-                                configurationId: agent.sId,
-                              },
-                            ]
-                          );
-                        }}
-                      >
-                        <AvatarListItem agent={agent} />
-                      </a>
-                    ))}
+        <GenerationContextProvider>
```

Comment:
You can probably put it only around FixedInputBar?

Or even move it down in FixedInputBar around InputBar and around AgentMessage in Conversation?

Reply:
I would prefer keeping only one provider instance, I even think I should have wrapped the whole AppLayout so that we always have Context Providers top levels of a page component -> centralizing the providers top-level to simplify the component trees, make them easily discoverable and make the code easier to maintain.

If you strongly feel it's better to have them duplicated in the tree I can add a commit tho 🙇🏻‍♀️

Reply:
I added a commit to put them top level for both new and /id